### PR TITLE
- Add A_OverlayTranslation to set PSprite translations

### PR DESCRIPTION
**Pull request by nashmuhandes**
- Added PSPF_PLAYERTRANSLATED support to the software renderer

### CODE REVIEW

**Comment by MajorCooke** [comment]
@coelckers , if you can, go ahead and merge this one first so I can set mine in order to his. It'll be a lot easier this way than the other way I think.

**Comment by nashmuhandes** [comment]
Forgot to add example file
[OverlayTranslationDemo.zip](https://github.com/coelckers/gzdoom/files/5333741/OverlayTranslationDemo.zip)

**Comment by madame-rachelle** [comment]
The potential pitfall I see here is truecolor sprites and truecolor models, as long as those continue to work (both of these need to be tested) with this change then this is safe to merge.

**Comment by nashmuhandes** [comment]
@madame-rachelle this simply uses code that already exists before I touched it. It was possible to set a weapon sprite to follow the player's color. I simply expanded it to allow the modder to specify a specific translation rather than just the player color.

Meaning, whatever problems with true color sprites/model skins would have existed in the engine regardless of this PR. And I believe those never worked correctly across the board in the engine, even on world sprites.

In other words, I believe true color sprites/model textures were never designed to work correctly with translations ever, no matter where the use case is.

**Comment by madame-rachelle** [comment]
I am not talking about translating truecolor graphics - I am only worried about them being glitched (if untranslated) after the merge. If they are still fine, then I am fine. :)

**Comment by nashmuhandes** [comment]
> I am not talking about translating truecolor graphics - I am only worried about them being glitched (if untranslated) after the merge. If they are still fine, then I am fine. :)

@madame-rachelle very well, here's evidence that true color weapon sprites did not break. It's fine. Safe to merge. :) https://imgur.com/a/rleNQYR

**Comment by MajorCooke** [comment]
And now I will update my PR for this one.

**Comment by Blue-Shadow** [comment]
Looking at the source, it doesn't look like this function works with predefined translations, like A_SetTranslation. Doesn't that limit the function's use, if the only way to set the translation of an overlay is by getting it from another actor (like how the demo does it)?

**Comment by nashmuhandes** [comment]
> Looking at the source, it doesn't look like this function works with predefined translations, like A_SetTranslation. Doesn't that limit the function's use, if the only way to set the translation of an overlay is by getting it from another actor (like how the demo does it)?

@Blue-Shadow I can add ``name`` support later, after I am sure @MajorCooke is done with his overlay-related PRs
